File: swarmmanagercontroller/src/main/java/co/uk/swarmbit/docker/api/swarm/SwarmApi.java
package co.uk.swarmbit.docker.api.swarm;

import co.uk.swarmbit.docker.api.common.json.UnlockKeyJson;
import co.uk.swarmbit.docker.api.swarm.parameters.SwarmUpdateParameters;
import co.uk.swarmbit.docker.api.common.json.SwarmJson;

public interface SwarmApi {

    SwarmJson inspectSwarm(String swarmId);

    void updateSwarm(String swarmId, SwarmUpdateParameters parameters);

    UnlockKeyJson unlock(String swarmId);
}

File: swarmmanagercontroller-new/src/main/kotlin/com/swarmbit/docker/api/swarm/SwarmApi.kt
package com.swarmbit.docker.api.swarm

import com.swarmbit.docker.api.swarm.model.Swarm
import com.swarmbit.docker.api.swarm.model.UnlockKey
import com.swarmbit.docker.api.swarm.parameters.SwarmUpdateParameters

interface SwarmApi {
    fun inspectSwarm(swarmId: String): Swarm
    fun updateSwarm(swarmId: String, parameters: SwarmUpdateParameters)
    fun unlock(swarmId: String): UnlockKey
}